File: mediq-backend/src/main/java/com/mediq/service/NotificationService.java
package com.mediq.service;

import com.mediq.dto.SmsNotificationRequest;
import com.mediq.dto.SmsNotificationResponse;

public interface NotificationService {

    SmsNotificationResponse sendRegistrationSms(Long patientId, Long campId);

    SmsNotificationResponse sendPrescriptionSms(Long prescriptionId);

    SmsNotificationResponse sendCustomSms(SmsNotificationRequest request);

    SmsNotificationResponse sendWhatsApp(SmsNotificationRequest request);
}

File: mediq-frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
  sendRegistrationSms: async (patientId, campId) =>
    api.post('/notifications/send-registration-sms', null, { params: { patientId, campId } }),

  sendPrescriptionSms: async (prescriptionId) =>
    api.post(`/notifications/send-prescription-sms/${prescriptionId}`),

  sendCustomSms: async (data) =>
    api.post('/notifications/send-custom-sms', data),

  sendWhatsApp: async (data) =>
    api.post('/notifications/send-whatsapp', data),
};

export default notificationService;
